README.md

```diff
@@ -113,80 +117,119 @@ However, it's a totally independent and standalone implementation (you don't nee
 
 ### Serving the Fragments
 
-To serve the fragments, React ESI provides a ready to use controller compatible with [Express](https://expressjs.com/):
+To serve the fragments, React ESI provides a ready to use controller compatible with [Express](https://expressjs.com/), you can find the full example [here](https://github.com/dunglas/react-esi/tree/main/examples/express)
 
 ```javascript
-// server.js
-import express from 'express';
-import { path, serveFragment } from 'react-esi/lib/server';
+// server.jsx
+import express from "express";
+import { path, serveFragment } from "react-esi/lib/server";
+import { renderToString } from "react-dom/server";
+import { App } from "./pages/App";
+import React from "react";
+
+const port = Number.parseInt(process.env.PORT || "3000", 10);
 
 const server = express();
 server.use((req, res, next) => {
   // Send the Surrogate-Control header to announce ESI support to proxies (optional with Varnish, depending of your config)
-  res.set('Surrogate-Control', 'content="ESI/1.0"');
+  res.set("Surrogate-Control", 'content="ESI/1.0"');
   next();
 });
 
-server.get(path, (req, res) =>
-  // "path" default to /_fragment, change it using the REACT_ESI_PATH env var
-  serveFragment(
+server.get("/", (req, res) => {
+  const app = renderToString(<App />);
+
+  const html = `
+      <html lang="en">
+      <head>
+          <script src="app.js" async defer></script>
+      </head>
+      <body>
+          <div id="root">${app}</div>
+      </body>
+      </html>
+  `;
+  res.send(html);
+});
+
+// "path" default to /_fragment, change it using the REACT_ESI_PATH env var
+server.get(path, (req, res) => {
+  return serveFragment(
     req,
     res,
     // "fragmentID" is the second parameter passed to the "WithESI" HOC, the root component used for this fragment must be returned
-    fragmentID => require(`./components/${fragmentID}`).default) 
-);
+    (fragmentID) => require(`./components/${fragmentID}`).default
+  );
+});
 
 // ...
 // Other Express routes come here
 
-server.listen(80);
+server.use(express.static("./dist"));
+
+server.listen(port, () => {
+  console.log(`> Ready on http://localhost:${port}`);
+});
 ```
 
-Alternatively, here is a full example using [a Next.js server](https://github.com/zeit/next.js/tree/master/examples/custom-server-express):
+Alternatively, here is a full example using [a Next.js server](https://github.com/dunglas/react-esi/tree/main/examples/next):
 
 ```javascript
```

Comment:
Are you suggesting that we should drop express in the example and stick with the http module like in the next documentation?

Reply:
Yes, or have both? I'm not sure :)

Reply:
I'd prefer to have both, but I'd like to extract the common code between examples (mainly react components) to avoid having to duplicate it, and just have a different server.ts file.
So perhaps we can defer these changes to another PR?

Reply:
No problem to defer. I'm not sure it is worth it to remove duplication in examples. It's better if examples are self-contained and easy to read.